src/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -1142,7 +1747,9 @@ public int CompareTo(object target)
             const int retInvalidEnumType = 3; // indicates that the enum was of an unknown/unsupported underlying type
 
             if (this == null)
```

Comment:
Anyone know what this is about? Is this really necessary? Looks like it's been in here since the coreclr repo was created.

Reply:
Call back from native code?

Reply:
Hmm, interesting. I guess we should validate that this method is still being called from native code and if not then remove the check?

Reply:
Unlike C#, IL does not prevent you from calling a method with null this pointer. Accessing the null this pointer in managed code produces NullReferenceException that the caller can handle like any other exception.

Accessing null this pointer in the unmanaged runtime causes immediate fatal crash that does not produce NullReferenceException. These explicit checks for null before calling unmanaged runtime call are there to still throw NullReferenceException instead of the fatal crash.